### ISSUE TITLE
CCRIndexLifecycleIT failures

### ISSUE THREAD

**Issue by joegallo** [elastic/elasticsearch#67546]
**Build scan**: https://gradle-enterprise.elastic.co/s/whuwnheb4mfdc (my machine)

**Repro line**: ``` ./gradlew ':x-pack:plugin:ilm:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testCannotShrinkLeaderIndex" -Dtests.seed=E80A68C2E753C2BC -Dtests.security.manager=true -Dtests.locale=cs-CZ -Dtests.timezone=America/Aruba -Druntime.java=15``` (note: this is just one example failure, it seems like there are many different ones from this class)

**Reproduces locally?**:  Yes, also pretty dependable for me with just `./gradlew :x-pack:plugin:ilm:check`.

**Applicable branches**: `master`

**Failure history**: https://build-stats.elastic.co/goto/923393018e49b47464700f864dca6ec8

**Failure excerpt**:
```
org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT > testCannotShrinkLeaderIndex FAILED
    java.lang.AssertionError: expected:<200> but was:<404>
        at __randomizedtesting.SeedInfo.seed([E80A68C2E753C2BC:4B5FCA46448679DC]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.lambda$testCannotShrinkLeaderIndex$26(CCRIndexLifecycleIT.java:460)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:928)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testCannotShrinkLeaderIndex(CCRIndexLifecycleIT.java:457)
```

**Comment by joegallo** [comment]
Some of the other failures I'm seeing from this class (again, locally):

```
> Task :x-pack:plugin:ilm:qa:multi-cluster:follow-cluster

REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testAliasReplicatedOnShrink" -Dtests.seed=E80A68C2E753C2BC -Dtests.security.manager=true -Dtests.locale=cs-CZ -Dtests.timezone=America/Aruba -Druntime.java=15

org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT > testAliasReplicatedOnShrink FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([E80A68C2E753C2BC:444B4BB635CA98F1]:0)
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.lambda$testAliasReplicatedOnShrink$15(CCRIndexLifecycleIT.java:335)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:928)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testAliasReplicatedOnShrink(CCRIndexLifecycleIT.java:335)

REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testILMUnfollowFailsToRemoveRetentionLeases" -Dtests.seed=E80A68C2E753C2BC -Dtests.security.manager=true -Dtests.locale=cs-CZ -Dtests.timezone=America/Aruba -Druntime.java=15

org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT > testILMUnfollowFailsToRemoveRetentionLeases FAILED
    java.lang.AssertionError:
    Expected: "complete"
         but: was "unfollow"
        at __randomizedtesting.SeedInfo.seed([E80A68C2E753C2BC:E35600926600EA94]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.assertILMPolicy(CCRIndexLifecycleIT.java:724)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.lambda$testILMUnfollowFailsToRemoveRetentionLeases$31(CCRIndexLifecycleIT.java:544)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:928)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testILMUnfollowFailsToRemoveRetentionLeases(CCRIndexLifecycleIT.java:543)

REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testCannotShrinkLeaderIndex" -Dtests.seed=E80A68C2E753C2BC -Dtests.security.manager=true -Dtests.locale=cs-CZ -Dtests.timezone=America/Aruba -Druntime.java=15

org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT > testCannotShrinkLeaderIndex FAILED
    java.lang.AssertionError: expected:<200> but was:<404>
        at __randomizedtesting.SeedInfo.seed([E80A68C2E753C2BC:4B5FCA46448679DC]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.lambda$testCannotShrinkLeaderIndex$26(CCRIndexLifecycleIT.java:460)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:928)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testCannotShrinkLeaderIndex(CCRIndexLifecycleIT.java:457)

REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testUnfollowInjectedBeforeShrink" -Dtests.seed=E80A68C2E753C2BC -Dtests.security.manager=true -Dtests.locale=cs-CZ -Dtests.timezone=America/Aruba -Druntime.java=15

org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT > testUnfollowInjectedBeforeShrink FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([E80A68C2E753C2BC:E0A347C8DD250AC1]:0)
        at org.junit.Assert.fail(Assert.java:86)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertTrue(Assert.java:52)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.lambda$testUnfollowInjectedBeforeShrink$22(CCRIndexLifecycleIT.java:390)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:928)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testUnfollowInjectedBeforeShrink(CCRIndexLifecycleIT.java:390)

REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testBasicCCRAndILMIntegration" -Dtests.seed=E80A68C2E753C2BC -Dtests.security.manager=true -Dtests.locale=cs-CZ -Dtests.timezone=America/Aruba -Druntime.java=15

org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT > testBasicCCRAndILMIntegration FAILED
    java.lang.AssertionError:
    Expected: "warm"
         but: was "hot"
        at __randomizedtesting.SeedInfo.seed([E80A68C2E753C2BC:AF0E17773D78FF32]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.assertILMPolicy(CCRIndexLifecycleIT.java:721)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.assertILMPolicy(CCRIndexLifecycleIT.java:709)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.lambda$testBasicCCRAndILMIntegration$2(CCRIndexLifecycleIT.java:97)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:928)
        at org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT.testBasicCCRAndILMIntegration(CCRIndexLifecycleIT.java:91)
```

**Comment by joegallo** [comment]
I'm going to try to `git bisect` this one, I don't think it used to fail on my machine.

**Comment by firstprayer** [comment]
Just ran this locally and it's passing now.

**Comment by gaobinlong** [comment]
Reproduces on my computer all the time, it may because the setting `indices.lifecycle.poll_interval` defaults to `10m`, and the injected `Unfollow Action` before `Rollover Action` in hot phase will stuck in `wait-for-follow-shard-tasks` step in the first round:

```
[2021-02-06T10:23:38,288][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [null] to [{"phase":"new","action":"complete","name":"complete"}] in policy [my_policy]
[2021-02-06T10:23:41,987][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"new","action":"complete","name":"complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] in policy [my_policy]
[2021-02-06T10:23:47,031][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"wait-for-indexing-complete"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] in policy [my_policy]
[2021-02-06T10:32:46,291][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"wait-for-follow-shard-tasks"}] to [{"phase":"hot","action":"unfollow","name":"pause-follower-index"}] in policy [my_policy]
[2021-02-06T10:32:46,437][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"pause-follower-index"}] to [{"phase":"hot","action":"unfollow","name":"close-follower-index"}] in policy [my_policy]
[2021-02-06T10:32:46,555][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"close-follower-index"}] to [{"phase":"hot","action":"unfollow","name":"unfollow-follower-index"}] in policy [my_policy]
[2021-02-06T10:32:46,687][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"unfollow-follower-index"}] to [{"phase":"hot","action":"unfollow","name":"open-follower-index"}] in policy [my_policy]
[2021-02-06T10:32:46,807][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"open-follower-index"}] to [{"phase":"hot","action":"unfollow","name":"wait-for-index-color"}] in policy [my_policy]
[2021-02-06T10:32:46,939][INFO ][o.e.x.i.IndexLifecycleTransition] [node1] moving index [my_index-1] from [{"phase":"hot","action":"unfollow","name":"wait-for-index-color"}] to [{"phase":"hot","action":"rollover","name":"check-rollover-ready"}] in policy [my_policy]

```

**Comment by martijnvg** [comment]
I'm not able to reproduce these failures locally. In CI I did see that in the last 30 days there were 4 failures regarding warning headers being returned. This is because legacy templates is being used. I will change the test to use composable index templates.

I also noticed that the ilm policy in the follower cluster also uses a rollover action in the hot phase. Given that the rollover occurs in the leader cluster, I think the ilm policy in the follower cluster shouldn't contain a rollover action in the hot phase, but only an unfollow action. I'm not sure whether that is related to the failure here.
